packages/back-end/src/services/segments.ts

```diff
@@ -6,7 +6,7 @@ export function toSegmentApiInterface(segment: SegmentInterface): ApiSegment {
     id: segment.id,
     name: segment.name,
     owner: segment.owner || "",
-    identifierType: segment.userIdType || "user_id",
```

Comment:
So when `toSegmentApiInterface` was created, it was done with some deviations - the SegmentInterface uses: `userIdType`, `datasource`, and `sql` properties instead of `identifierType`, `datasourceId`, and `query` properties.

I was running into a lot of issues with the `identifierType` property, so I've adjusted it here. But I'm pretty hesitant since our API isn't versioned.

It's also really odd to me that the shape of the data the REST API returns is different from that with which the user needs to pass in. But again, hesitant to change because our API isn't versioned.

@bryce-fitzsimons Curious if you have an opinion on the best way to go?

Reply:
Regarding the fallback value ("user_id"), probably best to leave it in for legacy purposes, unless we want to write a migration.

We chatted about renaming fields, and unfortunately are probably stuck with the schema that currently exists. Something to consider for API v2 perhaps...